gemini/use-cases/education/resume_generator.ipynb

```diff
@@ -0,0 +1,877 @@
+        "    \"\"\"\n",
+        "    model_response_extraction = model.generate_content(\n",
+        "        PROMPT_EXTR % (parsed_res),\n",
+        "    ).text\n",
+        "    model_response_gen = model.generate_content(\n",
+        "        PROMPT_GEN % (parsed_res, parsed_jd),\n",
+        "    ).text\n",
+        "    model_response_arr = model.generate_content(\n",
+        "        PROMPT_ARR % (parsed_res, parsed_jd),\n",
+        "    ).text\n",
+        "\n",
+        "    resume_generated = literal_eval(\n",
+        "        model_response_extraction.split(\"```json\")[-1].split(\"```\")[0].replace(\"\\n\", \"\")\n",
+        "    )\n",
+        "    resume_generated.update({\"Professional_Summary\": model_response_gen})\n",
+        "    resume_generated_3 = literal_eval(\n",
+        "        model_response_arr.split(\"```json\")[-1].split(\"```\")[0].replace(\"\\n\", \"\")\n",
+        "    )\n",
+        "    resume_generated.update(resume_generated_3)\n",
+        "\n",
+        "    return resume_generated"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "7924162c-7318-4c56-a673-a7411e6fdfd8"
+      },
+      "outputs": [],
+      "source": [
+        "def structure_response(resume_generated: str) -> str:\n",
+        "    \"\"\"\n",
+        "    Structure the resume to align with a given structure.\n",
+        "\n",
+        "    Args:\n",
+        "        resume_generated: The resume generated using LLM\n",
+        "\n",
+        "    Returns:\n",
+        "        The structured resume as string\n",
+        "    \"\"\"\n",
+        "    resume_structured = {}\n",
+        "    for k in resume_structure:\n",
+        "        resume_structured[k] = resume_generated[k]\n",
+        "    response_str = \"\"\n",
+        "    for block, block_info in resume_structured.items():\n",
+        "        response_str = f\"{response_str}\\n{block}\\n\"\n",
+        "        if block == \"Personal_Information\":\n",
+        "            tmp = \"\"\n",
+        "            for block_title, block_desc in block_info.items():\n",
+        "                tmp = f\"{tmp}\\n{block_title}: {block_desc}\\n\"\n",
+        "            response_str = f\"{response_str}\\n{tmp}\\n\"\n",
+        "        elif isinstance(block_info, list):\n",
+        "            # pylint: disable=broad-exception-caught\n",
+        "            try:\n",
+        "                response_str += \"\\n\".join(block_info) + \"\\n\"\n",
+        "            except:\n",
+        "                response_str += (\n",
+        "                    \"\\n\".join([f\"{k}:{v}\" for k, v in block_info[0].items()]) + \"\\n\"\n",
+        "                )\n",
+        "            # pylint: enable=broad-exception-caught\n",
+        "        else:\n",
+        "            response_str = f\"{response_str}\\n{block_info}\\n\"\n",
+        "    return response_str"
```

Comment:
This code could be simpler if you used a template like Jinja.

Reply:
Simplified the code